moveit_core/collision_detection_bullet/src/collision_env_bullet.cpp

```diff
@@ -187,8 +189,30 @@ void CollisionEnvBullet::checkRobotCollisionHelper(const CollisionRequest& req,
         cow->getName(), state.getAttachedBody(cow->getName())->getGlobalCollisionBodyTransforms()[0]);
   }
 
+  // Specify active collision links
+  if (!req.group_name.empty())
+  {
+    auto joint_model_group = robot_model_->getJointModelGroup(req.group_name);
+    std::vector<std::string> active_links = joint_model_group->getLinkModelNames();
```

Comment:
```suggestion
    const std::vector<std::string> active_links = joint_model_group->getUpdatedLinkModelsNames();
```
https://github.com/ros-planning/moveit/blob/74b3e30db2e8683ac17b339cc124675ae52a5114/moveit_core/collision_detection_fcl/src/collision_common.cpp#L934

https://github.com/ros-planning/moveit/blob/74b3e30db2e8683ac17b339cc124675ae52a5114/moveit_core/robot_model/include/moveit/robot_model/joint_model_group.h#L233

Reply:
Well, it stops detecting collisions with the yellow cube when I apply that change    :/

Reply:
That's weird as this almost follows the [enableGroup()](https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_common.cpp#L931) implementation used for [FCL](https://github.com/ros-planning/moveit/blob/master/moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp#L294). How about using `getUpdatedLinkModelsSet()` and get the names from there? That's actually how it works for FCL, not sure if there is something odd going on with duplicate entries.